src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import task.Deadline;
+import task.Event;
+import task.TaskList;
+import task.Todo;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the manner in which the program is parsed
+ * and expected to behave when user inputs are entered.
+ */
+public class Parser {
+
+
+    /**
+     * Processes the TaskList according to the input
+     * entered by the user.
+     *
+     * @param tasklist TaskList to process on
+     * @throws IOException If unexpected input is entered.
+     */
+    public void processData(TaskList tasklist) throws IOException {
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        String cmd = br.readLine();
```

Comment:
Maybe a more descriptive name can be used? e.g. command. I noticed this on some other parts of the code too